tests/system/workspace_factory.py

```diff
@@ -51,6 +51,21 @@ class Package(object):
     <build_type>{build_type}</build_type>
   </export>"""
 
+        CATKIN_CMAKELISTS_TEMPLATE = """
+cmake_minimum_required(VERSION 2.8.3)
+project({name})
+find_package(catkin REQUIRED COMPONENTS {catkin_components})
+catkin_package()"""
```

Comment:
There's a simplifying assumption here that a factory catkin package only depends on other catkin packages. All current tests pass with this assumption in place, so I see no reason to over complicate things.